src/iso7816.h

```diff
@@ -27,14 +27,13 @@ struct iso7816_header {
 	uint8_t lc3;
 })
 
-PACKED_TYPE(iso7816_apdu_t,
-struct iso7816_apdu {
+typedef struct iso7816_apdu {
 	size_t            alloc_len;
 	uint16_t          payload_len;
 	uint8_t          *payload_ptr;
 	iso7816_header_t  header;
 	uint8_t           payload[];
-})
+} iso7816_apdu_t;
```

Comment:
On platforms where `size_t` is 4 or 8 bytes, this causes one byte of padding to be added to the end of `iso7816_apdu_t`, which breaks the calculation in `iso7816_len()`. I suggest accounting for this extra byte when we set `alloc_len` in `iso7816_new()`.

Reply:
Ah indeed, and it's even greater on CHERI as our pointers on 64-bit architectures are 128-bit, so we end up with 9 padding byte.

Reply:
Looking at the resulting diff, it strikes me as simpler to (potentially) overallocate in `iso7816_new()` and have the adjustment logic entirely in `iso7816_len()`. I've just pushed a commit to that effect. Could I kindly ask you to squash it with yours? Provided you agree it's a sensible approach, of course.

Reply:
Yeah that's fine by me, overallocates a bit but is simpler to reason about. Also means `iso7816_len` is now always precise again, whereas my version would report it was longer than needed for small messages; don't know if that would have been a problem.